### DISCUSSION TITLE
Developers community

### DISCUSSION THREAD

**Discussion by MostHated** [nxrighthere/UnrealCLR#5] [General]
Hey man,
   First off, this seems great. I really had no plans to do anything with Unreal aside from play around with UE5 when it comes out, but after seeing this, that might change. I was wondering if you happen to already have, or plan on having a Discord server or similar setup for folks to discuss this project? I am a big fan of going back through past questions and technical discussions, trial and errors, etc. It definitely helps with something like this to have other like-minded devs to share info with without having to clog up the GitHub issues.

Thanks,
-MH

**Comment by nxrighthere** [comment]
While Discord is a straightforward and very convenient way to communicate with people, it falls apart when it comes to information searching and knowledge reuse. I'm maintaining several small communities at the moment and observing these problems for quite a while.

Give me some time to think about this more carefully.

**Comment by XeonG** [comment]
Looks good for Unreal, though very curious as to what they are cooking up for UE5. I'm done with supporting or helping Unity in any shape or form as I have done years ... there are some serious problems going on there, I would encourage anyone with a brain to avoid that engine for any future project or addon... its all going down (and publicly traded soon, a move by higher ups looking to cash out while the goings good, I can say right now the deplorable cancerous SJW rot has set in big time and its showing), its falling apart everywhere.

Anyway I do think a traditional threaded forum works best as a primary way for building a community and not discord(though useful aswel but more important to get users creating forum threads for support, requests etc) as it just can't be searched and discussions get buried so quickly (even github discussions get burried or closed), making it harder to pickup alot of things going on with the project/software. 

Have been looking at Cryengine/Unigine and now with this project, maybe Unreal, the key is mainly C# support, Unreal has lot of addons where as CryEngine/Unigine really don't right now but again C# is my preference hopefully this project can get further than the other C# supporting Unreal past projects have done.

**Comment by MostHated** [comment]
I definitely agree that a standard forum is pretty necessary for numerous reasons. Discord is mostly a personal preference in addition to a forum. I find it quite helpful to be in an active discord so I can, more or less, passively gain insight into general info, the issues folks are having, what the resolutions are, etc by seeing the ongoing discussions via notifications as I am doing other things, especially for newly released things like this.

**Comment by Tanner555** [comment]
You guys can always join the USharp gitter chat if you like. https://gitter.im/USharp/Lobby

Nxrighthere has recently joined the chat and has answered many of my questions. But I'd really like an official Gitter/Discord/Forum for UnrealCLR in the future. The community is growing rapidly.

**Comment by nxrighthere** [comment]
So here's the plan:
I'm currently working on core and workflow improvements based on feedback that I'm receiving from folks who are working on commercial projects, Meanwhile, I'm waiting for access to the new [GitHub Discussions](https://github.blog/2020-05-06-new-from-satellite-2020-github-codespaces-github-discussions-securing-code-in-private-repositories-and-more/#discussions) feature (see the example [here](https://github.com/vercel/next.js/discussions)), so that thing will be used for community discussions and questions. I'm not expecting wide adoption of the plugin in real-world projects (at least until the stable version release), so I think that should work just fine.

**Comment by MostHated** [comment]
I do hope that they enable this for the repo. I usually end up learning the most going through the back and forth of peoples discussions. I finally got everything installed and setup today and have been looking over the test files. I was excited to see just how familiar every ends up being in comparison to Unity. It seems like it should be a pretty smooth transition once I get a bit more familiar with the Unreal API offerings. 

I certainly understand how busy you must in keeping up with all of this, so I definitely don't want to take away from that but are there any plans for additional examples/test scenarios down the road with any more advanced things? 

I am not completely sure yet how it all works under the hood, but I am quite interested in seeing the comparative performance, such as something in UCLR vs the same written in C++, but also UCLR vs a similar scenario in Unity. I was always a fan of your networking library benchmarks from a while back. 

That said, it will be nice if the Discussions feature has, or eventually has, a way to compile together different contributions, such as examples and benchmarks and things of that nature that users have made. It could really add a lot of value, and make things much more "community" like in general, which would be quite beneficial to everyone involved. 

Keep up the great work, man. 👍 

(P.S. have you looked at all into the grants that Epic offers? This seems like exactly the kind of project for which that was started in the first place.)

**Comment by nxrighthere** [comment]
> I certainly understand how busy you must in keeping up with all of this, so I definitely don't want to take away from that but are there any plans for additional examples/test scenarios down the road with any more advanced things?

Yes, I'm currently working on a full-featured game demo since some people are confused about how to use the plugin optimally after traditional paradigms.

> I am not completely sure yet how it all works under the hood, but I am quite interested in seeing the comparative performance, such as something in UCLR vs the same written in C++, but also UCLR vs a similar scenario in Unity. I was always a fan of your networking library benchmarks from a while back.

It's hard to objectively compare performance in this case, but I'll take a look what I can do.

> (P.S. have you looked at all into the grants that Epic offers? This seems like exactly the kind of project for which that was started in the first place.)

Yes, the application review in progress.

> Keep up the great work, man. 👍

Thanks!

**Comment by MostHated** [comment]
Awesome, all great news. I appreciate the reply.

**Comment by icavalheiro** [comment]
I just wanted to come by and say a big THANK YOU to @nxrighthere for this wonderful initiative.
This might make me switch from Unity to Unreal, this is just the last part missing for me.

Bro, thank you.

**Comment by nxrighthere** [comment]
@icavalheiro I'm glad that you like it! There is still some work to do, but we will be there eventually.

**Comment by nxrighthere** [comment]
We got [discussions](https://github.com/nxrighthere/UnrealCLR/discussions) for the project, thanks to GitHub staff and @becca in particular! ❤️

**Reply by becca** [reply]
Here to help! Excited to see how this helps you unify the community and all its knowledge. 💖

**Reply by nxrighthere** [reply]
@becca Can you help me, please? I can't figure out how to set up categories like [here](https://github.com/vercel/next.js/discussions).

**Reply by becca** [reply]
sure thing! just enabled them for you 🚀

**Reply by nxrighthere** [reply]
Awesome, thanks!

**Comment by MostHated** [comment]
Awesome news. 👍
